RichEditorView/Classes/RichEditorView.swift

```diff
@@ -439,6 +445,75 @@ extension RichEditorView: UIGestureRecognizerDelegate {
 // MARK: - Utilities
 extension RichEditorView {
     
+    /** Sets content offset for scrollView based on caret location **/
+    
+    private func fixScrollView(editor: RichEditorView) {
+        
+        var scrollView = editor.webView.scrollView
+        
+        var contentHeight: CGFloat?
+        let htmlHeight = editor.runJS("document.getElementById('editor').clientHeight;")
+        if let n = NSNumberFormatter().numberFromString(htmlHeight) {
+            let floatValue = CGFloat(n)
+            contentHeight = floatValue
+        } else {
+            contentHeight = scrollView.frame.height
+        }
+        scrollView.contentSize = CGSizeMake(scrollView.frame.width, contentHeight!)
+        
+        /* Caret above scrollView bounds */
+        
+        // Usually getCaretPosition() returns the amount the caret is offset from the current contentOffset, so "-40" would
+        // indicate that the contentOffset would need to be decreased by 40.
+        // However, when the cursor is on a new line with no html content, getCaretPosition() returns the absolute position
+        // of the caret.  The resulting code differentiates between caret position values greater than 0 that could either mean
+        // the caret position is beyond the scrollView lower bounds, or above the scrollView upper bounds and on a new line.
+        // The code also checks if the caret position is within the scrollView bounds to avoid scrolling when caret is 
+        // not on the first line of the scrollView.
+        
+        let data = convertStringToDictionary(editor.getCaretPosition())
+        
+        // Checks if caret is on new line with no content
+        let newLine = data!["newLine"] as! Bool
+        if let caretPositionNumeric = data!["height"] as? NSNumber {
```

Comment:
so a few things - `height` and `newLine` aren't x and y values. The x value of the caret is irrelevant for vertical scrolling. `newLine` indicates whether the caret is on a new line or not, and therefore which branch of the conditional to execute.  The only way I could return those two values was using a dictionary, because it seemed from the other JS that the return value had to be a String.